src/utils/output-target.ts

```diff
@@ -37,7 +37,14 @@ export const relativeImport = (pathFrom: string, pathTo: string, ext?: string, a
 
 export const getComponentsDtsSrcFilePath = (config: d.ValidatedConfig) => join(config.srcDir, GENERATED_DTS);
 
-export const getComponentsDtsTypesFilePath = (outputTarget: d.OutputTargetDist | d.OutputTargetDistTypes) =>
+/**
+ * Helper to get an appropriate file path for `components.d.ts` for a `"dist"`
+ * or `"dist-types"` output target.
+ *
+ * @param outputTarget the output target of interest
+ * @returns a properly-formatted path
+ */
+export const getComponentsDtsTypesFilePath = (outputTarget: Required<d.OutputTargetDist> | d.OutputTargetDistTypes) =>
```

Comment:
I added the `Required<>` wrapper here to make sure the `.typesDir` is defined, and this is legit since `getComponentsDtsTypesFilePath` is only called in the `validate-dist` file above where the argument satisfies that type